cmd/kubelet/kubelet.go

```diff
@@ -32,29 +33,52 @@ import (
 	"k8s.io/kubernetes/cmd/kubelet/app"
 	"k8s.io/kubernetes/cmd/kubelet/app/options"
 	_ "k8s.io/kubernetes/pkg/client/metrics/prometheus" // for client metric registration
-	_ "k8s.io/kubernetes/pkg/version/prometheus"        // for version metric registration
+	"k8s.io/kubernetes/pkg/kubelet/apis/kubeletconfig"
+	_ "k8s.io/kubernetes/pkg/version/prometheus" // for version metric registration
 	"k8s.io/kubernetes/pkg/version/verflag"
 )
 
+func newFlagSet(kf *options.KubeletFlags, kc *kubeletconfig.KubeletConfiguration) *pflag.FlagSet {
```

Comment:
I think this would be easier to follow inline in `main()`. We want to make sure all logic is able to be used by other callers, and the KubeletFlags and KubeletOptions structs don't get entangled with each other... passing both into a private helper makes it harder to follow the logic and ensure the steps are sequential and distinct.

```
// init flagset with global options
fs := pflag.NewFlagSet(os.Args[0], pflag.ExitOnError)
fs.SetNormalizeFunc(flag.WordSepNormalizeFunc)
options.AddGlobalFlags(fs)

// bind to flag struct
kubeletFlags := options.NewKubeletFlags()
kubeletFlags.AddFlags(fs)

// bind to options struct
defaultConfig, err := options.NewKubeletConfiguration()
if err != nil {
  die(err)
}
options.AddKubeletConfigFlags(fs, defaultConfig)

// parse
if err := fs.Parse(os.Args[1:]); err != nil {
  die(err)
}
```

Reply:
I initially found the helper attractive, but to your point I've seen enough of these "pass all the big objects in and magic happens" helpers rot by new that I think I agree...

Reply:
Done